MediaGalleryUi/view/adminhtml/ui_component/product_listing.xml

```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+-->
+<listing xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Ui:etc/ui_configuration.xsd">
+    <listingToolbar name="listing_top">
+        <filters name="listing_filters">
+              <filterSelect
+                name="asset"
+                provider="${ $.parentName }"
+                sortOrder="10"
+                class="Magento\MediaGalleryUi\Ui\Component\Listing\Filters\Asset"
+                component="Magento_MediaGalleryUi/js/form/element/ui-select"
+                template="Magento_MediaGalleryUi/grid/filters/elements/ui-select">
+                <argument name="data" xsi:type="array">
+                    <item name="config" xsi:type="array">
+                        <item name="entityType" xsi:type="string">catalog_product</item>
+                        <item name="identityColumn" xsi:type="string">entity_id</item>
+                        <item name="chipsEnabled" xsi:type="boolean">true</item>
```

Comment:
Please add 

                        <item name="filterPlaceholder" xsi:type="string" translate="true">Asset Title</item>
                        <item name="emptyOptionsHtml" xsi:type="string" translate="true">Start typing to find assets</item>